drivers/SmartThings/jbl/src/discovery_mdns.lua

```diff
@@ -0,0 +1,154 @@
+local log = require "log"
+local mdns = require "st.mdns"
+local net_utils = require "st.net_utils"
+
+local discovery_mdns = {}
+
+local function byte_array_to_plain_text(byte_array)
```

Comment:
This function will only work if every byte is an ASCII byte. Are we sure that there won't be any UTF-8 payloads getting passed in to this?